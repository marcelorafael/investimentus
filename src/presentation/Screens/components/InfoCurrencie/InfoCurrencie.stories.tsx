import React from 'react';
import { text, withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import { View, Text } from 'react-native';
import { InfoCurrencie } from './InfoCurrencie'

storiesOf('Screens/InfoCurrencie', module)
  .addDecorator(withKnobs)
  .add('default', () => (
    <View style={{flex: 1, justifyContent:'flex-end'}}>
      <InfoCurrencie
        title={text('title','Dollar')} 
        buy={text('buy','255')}
        sell={text('sell','6.68')}
        variation={text('variation','-5.6')} 
      />
    </View>
  ))
