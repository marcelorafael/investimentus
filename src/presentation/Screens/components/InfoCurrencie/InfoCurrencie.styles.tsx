import styled from 'styled-components/native';
import LinearGradient from 'react-native-linear-gradient';

export const Container = styled.View`

justify-content: flex-end;
align-items: center;
`;

export const Header = styled.View`
width: 100%;

flex-direction: row;
align-items: center;
justify-content: space-between;

padding-left: 12px;
padding-right: 12px;

margin-top: 10px;
`;

export const CenterView = styled.View`
margin-top: 8px;
align-items: center;

padding-left: 12px;
padding-right: 12px;

flex-direction: row;
justify-content: space-between;
`;

export const Title = styled.Text`
${({theme}) => theme?.FONTS?.oswaldBold ?? {}}
color: ${({theme}) => theme?.COLORS?.white ?? '#FFF'};
text-align: center;
margin-top: 10px;
`;

export const TextCurrency = styled.Text`
${({theme}) => theme?.FONTS?.oswaldBold ?? {}}
color: ${({theme}) => theme?.COLORS?.white ?? '#FFF'};
text-align: center;
`;

export const TextCenter = styled.Text`
${({theme}) => theme?.FONTS?.oswald ?? {}}
color: ${({theme}) => theme?.COLORS?.white ?? '#FFF'};
`;

export const ImgMoney = styled.Image`
width: 180px;
height: 180px;

margin-bottom: 50px;
`;

export const FooterCard = styled(LinearGradient).attrs({
  colors:['#000','#7F5DF0']
})`
width: 90%;
padding: 10px;
margin: 10px;

border-radius: 5px;
`;