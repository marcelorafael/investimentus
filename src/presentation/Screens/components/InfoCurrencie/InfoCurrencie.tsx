import React from 'react'
import { View } from 'react-native';
import { icons } from '../../../../../constants';
import {
  Container, Header, CenterView, TextCurrency,
  Title, TextCenter, ImgMoney, FooterCard,
} from './InfoCurrencie.styles';

interface InfoCurrencieProps {
  title: string
  variation: string
  buy: string
  sell: string
}

const InfoCurrencie: React.FC<InfoCurrencieProps> = (
  props: InfoCurrencieProps
) => {
  const { buy, sell, variation, title } = props;
  return (
    <Container>
      <ImgMoney source={icons.money} />
      <FooterCard>
        <Title>{title}</Title>
        <Header>
          <TextCurrency>R$ = {buy}</TextCurrency>
        </Header>
        <CenterView>
          <View>
            <TextCenter>Venda</TextCenter>
            <TextCenter>R$ = {sell}</TextCenter>
          </View>

          <View style={{ alignItems: 'flex-end' }}>
            <TextCenter>Variação</TextCenter>
            <TextCenter>{variation}</TextCenter>
          </View>
        </CenterView>
      </FooterCard>
    </Container>
  )
}
const InfoCurrencieMemo = React.memo(InfoCurrencie);
export { InfoCurrencieMemo as InfoCurrencie };