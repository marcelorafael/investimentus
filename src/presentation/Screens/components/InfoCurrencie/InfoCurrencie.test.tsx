import React from 'react'
import renderer from 'react-test-renderer'
import { InfoCurrencie } from './InfoCurrencie'

describe('Screens/InfoCurrencie', () => {
  it('should match snapshot', () => {
    const sut = <InfoCurrencie title="" buy="" sell="" variation="" />

    const renderComponent = renderer.create(sut)

    expect(renderComponent).toMatchSnapshot()
  })
})
