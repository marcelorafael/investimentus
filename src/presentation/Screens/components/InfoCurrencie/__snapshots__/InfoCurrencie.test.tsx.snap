// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screens/InfoCurrencie should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "flex-end",
      },
    ]
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../assets/icons/money.png",
      }
    }
    style={
      Array [
        Object {
          "height": 180,
          "marginBottom": 50,
          "width": 180,
        },
      ]
    }
  />
  <BVLinearGradient
    colors={
      Array [
        4278190080,
        4286537200,
      ]
    }
    endPoint={
      Object {
        "x": 0.5,
        "y": 1,
      }
    }
    locations={null}
    startPoint={
      Object {
        "x": 0.5,
        "y": 0,
      }
    }
    style={
      Array [
        Object {
          "borderRadius": 5,
          "marginBottom": 10,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 10,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
          "width": "90%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFF",
            "marginTop": 10,
            "textAlign": "center",
          },
        ]
      }
    >
      
    </Text>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginTop": 10,
            "paddingLeft": 12,
            "paddingRight": 12,
            "width": "100%",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFF",
              "textAlign": "center",
            },
          ]
        }
      >
        R$ = 
        
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginTop": 8,
            "paddingLeft": 12,
            "paddingRight": 12,
          },
        ]
      }
    >
      <View>
        <Text
          style={
            Array [
              Object {
                "color": "#FFF",
              },
            ]
          }
        >
          Venda
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#FFF",
              },
            ]
          }
        >
          R$ = 
          
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#FFF",
              },
            ]
          }
        >
          Variação
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#FFF",
              },
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </BVLinearGradient>
</View>
`
